### ISSUE TITLE
Epic: Integrate Auth + Workspace Management API's into VS Code Desktop

### ISSUE THREAD

**Issue by loujaybee** [gitpod-io/gitpod#9757]
| Context             | Integrate the newly created workspace Authentication and workspace management API's [#8512](https://github.com/gitpod-io/gitpod/issues/8512) into VS Code Desktop. |
|---------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Owner               | Product - @loujaybee / Tech - @akosyakov                                                                                                                                                                                                                              |
| Value               | Enable Gitpod users to have a full desktop IDE experience which does not require the user to authenticate or navigate to Gitpod in the browser to start or manage workspaces  |
| Acceptance Criteria | - Auth flow is replaced with the new authentication API endpoints                                                                                                                                                                                            |
| Growth Area         | Activation & Retention / Expansion                                                                                                                                                                                                                                    |
| Persona(s)          | N/A                                                                                                                                                                                                                                                                   |
| Hypothesis          | N/A |
| Measurement         | N/A                                                                                                                                                                                                                                                                   |
| In scope            | N/A                                                                                                                                                                                                                                                                   |
| Out of scope        | |
| Complexities        | - Work is dependent on API's introduced as part of #7900                                                                                                                                                                                                                                                                  |
| Latest Update       | 28.04.2022 - Updated note about dependencies on public API's. We cannot start this epic until that work progresses further                                                                                                                                                                                           |

**Comment by loujaybee** [comment]
- We intend to scope this down and exclude auth + token related APIs to simplify
- We should consider scoping this down to the current version (version 1 - experimental)

**Comment by loujaybee** [comment]
@akosyakov to either: 

1. Re-scope this to integration into supervisor, not VS Code. 
2. Create another epic for supervisor, and move this issue to the backlog.
